### PR TITLE
Presize the buffer

### PR DESCRIPTION
**Pull request by ghost**
presizes the buffer to avoid re-allocations while padding.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -11,7 +11,7 @@ pub fn left_pad(s: &str, pad: u32) -> String
 /// pad a string to the left with ``pad`` ``padchar``
 pub fn left_pad_char(s: &str, pad: u32, padchar: char) -> String
 {
-    let mut out = String::new();
+    let mut out = String::with_capacity( s.as_bytes().len() + (pad as usize) );
```
Comment (hfiguiere) [inline src/lib.rs:14]: I thought about doing that. It happens that your patch is based of 0.1.0 while I already have 0.2.0 that declare `pad` as `usize`. I you can rebase, I'll be happy to merge.

**Comment by hfiguiere** [comment]
Thanks !

File: src/lib.rs (hunk 1 above)
Comment (hfiguiere) [inline src/lib.rs:14]: also the behaviour of the function was wrong and was fixed in 0.2.0, so you will have to fix that too.

**Comment by ConnyOnny** [comment]
This is wrong for more complicated pad chars.
It should be `s.as_bytes().len() + (pad as usize) * padchar.len_utf8()`

EDIT: And this misses a cmp::max in there. I'll create another PR...

EDIT2: Done. PR #6 supersedes this PR.

**Pull request closed without merging**